purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields
+from odoo.tests import SavepointCase
+
+
+class TestPurchaseOrderSecondaryUnit(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_uom_kg = cls.env.ref('uom.product_uom_kgm')
+        cls.product_uom_gram = cls.env.ref('uom.product_uom_gram')
+        cls.product_uom_unit = cls.env.ref('uom.product_uom_unit')
+        cls.product = cls.env['product.product'].create({
+            'name': 'test',
+            'uom_id': cls.product_uom_kg.id,
+            'uom_po_id': cls.product_uom_kg.id,
+            'secondary_uom_ids': [
+                (0, 0, {
+                    'name': 'unit-700',
+                    'uom_id': cls.product_uom_unit.id,
+                    'factor': 0.7,
+                })],
+        })
+        cls.secondary_unit = cls.env['product.secondary.unit'].search([
+            ('product_tmpl_id', '=', cls.product.product_tmpl_id.id),
+        ])
+        cls.product.purchase_secondary_uom_id = cls.secondary_unit.id
+        cls.partner = cls.env['res.partner'].create({
+            'name': 'test - partner',
+            'supplier': True,
+        })
+        po = cls.env['purchase.order'].new({
+            'partner_id': cls.partner.id,
+            'company_id': cls.env.user.company_id.id,
+            'order_line': [(0, 0, {
+                'name': cls.product.name,
+                'product_id': cls.product.id,
+                'product_qty': 1,
+                'product_uom': cls.product.uom_id.id,
+                'price_unit': 1000.00,
+                'date_planned': fields.Datetime.now(),
+            })],
+        })
+        po.onchange_partner_id()
+        cls.order = cls.env['purchase.order'].create(
```

Comment:
From Travis log, seem to have error in this line as following,
```
ERROR: null value in column "picking_type_id" violates not-null constraint
DETAIL:  Failing row contains (8, null, null, PO00008, null, null, 2019-04-15 04:24:45, null, 43, null, 2, draft, null, 0, no, null, null, null, null, null, null, null, 1, 1, 1, 2019-04-15 04:24:45.10274, 1, 2019-04-15 04:24:45.10274, null, null, null).

```

Reply:
@Nikul-Chaudhary I load this module and test locally. I found that,
If I only install **purchase_order_secondary_unit** and run test -> OK
But if I install **purchase_stock** and run test -> Error as above, due to field picking_type_id is introduced by the module.
And this is because **purchase_delivery_split_date** in this repo install it.

@pedrobaeza I also wonder, what to do in this kind of situation?

Reply:
You are not testing in local the same way as here in Travis. In Travis, all dependencies are first install in a step, and then modules from the repo are installed and tested at the same time. So you should imitate that way of testing. This is not a real error right now.

A workaround is to put tests to be run at post-install in local. We tried to normalize the situation in https://github.com/odoo/odoo/pull/27883, but it's not accepted right now.